sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3562,11 +3560,26 @@ object GpuOverrides extends Logging {
       // This needs to match what murmur3 supports.
       PartChecks(RepeatingParamCheck("hash_key",
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
-            TypeSig.STRUCT).nested(), TypeSig.all)),
+            TypeSig.STRUCT + TypeSig.ARRAY).nested(),
+        TypeSig.all)
+      ),
       (hp, conf, p, r) => new PartMeta[HashPartitioning](hp, conf, p, r) {
         override val childExprs: Seq[BaseExprMeta[_]] =
           hp.expressions.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
 
+        override def tagPartForGpu(): Unit = {
+          val arrayWithStructsHashing = hp.expressions.exists(e =>
+            TrampolineUtil.dataTypeExistsRecursively(e.dataType,
+              dt => dt match {
+                case ArrayType(_: StructType, _) => true
+                case _ => false
+              })
+          )
+          if (arrayWithStructsHashing) {
+            willNotWorkOnGpu("hashing arrays with structs is not supported")
```

Comment:
Is there a follow on issue to fix this? Have we tested that this does not work?

Reply:
@sameerz is hashing function for Array[Struct] supported in cudf or is there an issue tracking that?

Reply:
In cudf, lists of structs and structs of lists are not yet supported (tracking issue https://github.com/rapidsai/cudf/issues/11222).

Reply:
Also spark-rapid tracking issue: https://github.com/NVIDIA/spark-rapids/issues/5109

Reply:
@ttnghia, the cudf issue you have tagged rapidsai/cudf#11222 seems to be for sorting a list of structs, is that the same for hashing? I tested passing a list of structs as groupBy key and cudf didn't like it.

Reply:
If I understand what you are saying correctly, I am aware that we don't have any control in the plugin to pick which aggregation to use, cudf will pick Hash or Sort. 
But the line this comment is referencing is explicitly talking about `HashPartitioning` so I think this very explicitly only concerns that. To prove this, I uncommented this check and tried to do a groupBy on a List[Struct] and got an error from cudf saying that murmur hash is not implemented for List[Struct]. This is why I was confused why issue rapidsai/cudf#11222, which is completed, is tagged here when it deals with sorting a list and clearly doesn't fix the problem with groupBy on List[Struct]

Reply:
Then we have to answer the question about what does `HashPartitioning` do for groupby?

Reply:
If groupBy uses `HashAggregate` then it will use `HashPartitioning` to create buckets with buffers pointing to values. So in this case, it will try to calculate the hash for List[Struct] which isn't supported atm by cudf.

Reply:
I've filed an issue for it: https://github.com/NVIDIA/spark-rapids/issues/8676

Reply:
Thanks. Is there a corresponding cudf issue?